### PR TITLE
Add cast.incomparable signedness warning for testcases

### PR DESCRIPTION
**Pull request by AndrewShf**
These tests are all converting a signed object to an unsigned object, or the other way around, but the two qualifiers do not have a subtype relationship.

As shown by the error messages
```
./daikon/Quant.java:3405: error: [signedness:cast.incomparable] incomparable typecast from "@Signed Object" to "@Unsigned Character"
        return new char[] { ((Character)fieldObj).charValue() };
                             ^
./daikon/Quant.java:3494: error: [signedness:cast.incomparable] incomparable typecast from "@Signed Object" to "@Unsigned Character"
    return ((Character)fieldObj).charValue();
            ^
./daikon/chicory/PureMethodInfo.java:184: error: [signedness:cast.incomparable] incomparable typecast from "@Signed Object" to "@Unsigned Character"
      return new Runtime.CharWrap((Character) obj);

```

For these cases, we can just suppress the warning, and in general I think these new incomparable errors make sense in the signedness type system.

### CODE REVIEW

File: java/daikon/AnnotateNullable.java [hunk 1]
```diff
@@ -134,7 +134,8 @@ public static void main(String[] args) throws IOException {
       if (ppt.is_class()) {
         @SuppressWarnings(
             "nullness") // map: retrieve class name from class Ppt name, with string manipulation
-        @NonNull List<PptTopLevel> static_methods = class_map.get(ppt.name().replace(":::CLASS", ""));
+        @NonNull List<PptTopLevel> static_methods =
```
Comment (wmdietl) [inline java/daikon/AnnotateNullable.java]: This PR now contains many unrelated formatting changes. Can you undo these?

File: java/daikon/QuantBody.java.jpp [hunk 2]
```diff
@@ -2186,6 +2186,7 @@ The "defined(INT)" test ensures that it only appears once in the output.
     assert (fields != null);
     assert (fieldsStartIdx >= 0 && fieldsStartIdx < fields.length);
 
+    @SuppressWarnings("signedness:cast.incomparable")
```
Comment (wmdietl) [inline java/daikon/QuantBody.java.jpp]: Note that this suppression has no effect: it would only suppress something from this declaration, but that's not what you want here, as there is no cast.

File: java/daikon/chicory/PureMethodInfo.java [hunk 3]
```diff
@@ -181,7 +181,9 @@ public PureMethodInfo(
     } else if (obj instanceof Byte) {
       return new Runtime.ByteWrap((Byte) obj);
     } else if (obj instanceof Character) {
-      return new Runtime.CharWrap((Character) obj);
+      @SuppressWarnings("signedness:cast.incomparable")
```
Comment (wmdietl) [inline java/daikon/chicory/PureMethodInfo.java:184]: Can you add a comment why this can be suppressed?

**Review by wmdietl — CHANGES_REQUESTED** [review]
Thanks! Please also add a cross-reference to the corresponding eisop PR.

File: java/daikon/QuantBody.java.jpp (hunk 2 above)
Comment (AndrewShf) [reply java/daikon/QuantBody.java.jpp]: Thanks, already addressed.

File: java/daikon/AnnotateNullable.java (hunk 1 above)
Comment (AndrewShf) [reply java/daikon/AnnotateNullable.java]: sure.

File: java/daikon/chicory/PureMethodInfo.java (hunk 3 above)
Comment (AndrewShf) [reply java/daikon/chicory/PureMethodInfo.java:184]: added

**Comment by AndrewShf** [comment]
The fix for the daikon is for this PR: https://github.com/eisop/checker-framework/pull/337

**Comment by AndrewShf** [comment]
Now we don't need to this PR, as the logic in Signedness checker has changed. These typecastings are considered legal now in Signedness Checker.
